support/hololens/ServoApp/BrowserPage.cpp

```diff
@@ -219,9 +184,9 @@ void BrowserPage::OnHistoryChanged(bool back, bool forward) {
 }
 
 void BrowserPage::OnShutdownComplete() {
-  log("Servo notified ShutdownComplete");
-  HANDLE hEvent = ::OpenEventA(EVENT_ALL_ACCESS, FALSE, sShutdownEvent);
-  ::SetEvent(hEvent);
+  EnterCriticalSection(&mGLLock);
+  mLooping = false;
+  LeaveCriticalSection(&mGLLock);
```

Comment:
Here you might also need to add a `WakeConditionVariable(&mGLCondVar);`, to make sure the loop wakes-up and sees the change.

Reply:
That should not be necessary as we know that at the point that we are not waiting on the variable.

Reply:
I'm not sure I understand the flow then. 

If `mTasks.size() == 0 && !mAnimating && mLooping`, the loop might be waiting on the condvar. In that case it won't see the change of `mLooping = false` unless woken-up. 

And `RequestShutdown` doesn't seem to do anything to ensure it is not waiting when `OnShutdownComplete` is called.

Reply:
Since `RequestShutdown` is called at https://github.com/servo/servo/pull/23904/files#diff-9f1e08fa0c4bb110f99fabbfeb609314R40, which is when `mLooping`, so then when OnShutdownComplete is called by Servo, `mLooping` would still be true, then you set it to false, and for the loop to stop it would need wake-up and see the change to `!mLooping`?

Reply:
>  the loop might be waiting on the condvar

No because at that point we are running code in the compositor thread. `OnShutdownComplete` is called from `perform_update`. It's impossible to be waiting on the condvar at that point.

so after `OnShutdownComplete` is called, `perform_udpate` finishes, we get back to the beginning of the loop, `mLooping` is checked. And we good :)

Reply:
Yep I think you're right, inside `mServo->PerformUpdates()` it's set to false, and then you go back to the top and you don't wait, going straight to `break`. 

looks good.